packages/polaris-viz/src/components/LineChartRelational/components/CustomLegend/CustomLegend.tsx

```diff
@@ -48,31 +103,63 @@ export function CustomLegend({
               color={color!}
               index={index}
               isComparison={isComparison}
-              name={seriesNameFormatter(name ?? '')}
-              shape="Line"
+              name={seriesNameFormatter(metadata?.legendLabel ?? name)}
+              shape={styleOverride?.tooltip?.shape ?? 'Line'}
               lineStyle={metadata?.lineStyle}
               theme={theme}
+              onDimensionChange={onDimensionChange}
             />
           </li>
         );
       })}
-      <li
-        key={percentileIndex}
-        style={{
-          ...getColorVisionStyles(percentileIndex),
-        }}
-        {...getColorVisionEventAttrs(percentileIndex)}
-      >
-        <LegendItem
-          color={
-            percentileItems[0].color ?? percentileItems[0]?.metadata?.areaColor
-          }
-          index={percentileIndex}
-          name={seriesNameFormatter(percentileItems[0]?.metadata?.legendLabel)}
-          shape="Bar"
+
+      {!hasHiddenData && percentileItems.length > 0 && (
```

Comment:
Why not just having this line item removed and just iterate over the visible series? 🤔